storage2/src/storage.rs

```diff
@@ -29,16 +64,129 @@ impl Storage {
     }
 
     // TODO: this probably can't be 'static long-term
-    pub async fn apply(&'static mut self, state: State) {
-        // TODO: 1. write the index tables and JMT to RocksDB
-        // 2. update the snapshot
-        // TODO: set jmt_version correctly
-        let jmt_version = 0;
-        let snapshot = self.db.snapshot();
-        // Obtain the write-lock for the latest snapshot, and replace it with the new snapshot.
-        let mut guard = self.latest_snapshot.write();
-        *guard = Arc::new(Snapshot::new(snapshot, jmt_version));
-        // Drop the write-lock (this will happen implicitly anyways, but it's good to be explicit).
-        drop(guard);
+    pub async fn apply(&'static mut self, state: State, nct: &Tree) -> Result<()> {
```

Comment:
Does this need to take `self` as `&'static mut`, or can it just be `&mut`?

Reply:
If I remove the lifetime, the build breaks

Reply:
What's the error?

Reply:
```
borrowed data escapes outside of associated function
`self` escapes the associated function body here
```

referring to the tokio task

Reply:
Oh I see, yes, that makes sense. Tokio tasks can only take `'static` things, because they might outlive the scope in which they are spawned.